monai/transforms/transforms.py

```diff
@@ -80,6 +81,65 @@ def __call__(self, img):
         return np.flip(img, self.axis)
 
 
+@export
+class Zoom:
+    """ Zooms a 3d image. Uses scipy.ndimage.zoom or cupyx.scipy.ndimage.zoom in case of gpu. 
+    For details, please see https://docs.scipy.org/doc/scipy/reference/generated/scipy.ndimage.zoom.html.
+
+    Args:
+        zoom (float or sequence): The zoom factor along the axes. If a float, zoom is the same for each axis. 
+            If a sequence, zoom should contain one value for each axis.
+        order (int): order of interpolation. Default=3.
+        mode (str): Determines how input is extended beyond boundaries. Default is 'constant'.
+        cval (scalar, optional): Value to fill past edges. Default is 0.
+        use_gpu (bool): Should use cpu or gpu.
+        keep_size (bool): Should keep original size (pad if needed).
+    """
+    def __init__(self, zoom, order=3, mode='constant', cval=0, prefilter=True, use_gpu=False, keep_size=False):
+        assert isinstance(order, int), "Order must be integer."
+        self.zoom = zoom
+        self.order = order
+        self.mode = mode
+        self.cval = cval
+        self.prefilter = prefilter
+        self.use_gpu = use_gpu
+        self.keep_size = keep_size
+
+    def __call__(self, img):
+        zoomed = None
+        if self.use_gpu:
+            try:
+                import cupy
+                from cupyx.scipy.ndimage import zoom as zoom_gpu
+
+                zoomed_gpu = zoom_gpu(cupy.array(img), zoom=self.zoom, order=self.order,
+                                      mode=self.mode, cval=self.cval, prefilter=self.prefilter)
+                zoomed = cupy.asnumpy()
+            except Exception:
+                print('Warning: Zoom gpu failed. Defaulting to cpu.')
```

Comment:
i think here we choose gpu or cpu in the constructor, then we don't try and except in every call

Reply:
Is cupy an approved dependency for MONAI?

Reply:
@ericspod @wyli what do you think about cupy? I can totally remove it and leave it as "to-do" for later.

Reply:
i feel we don't put cupy in `requirements.txt`,  but leave the cupy calls here. if the user set `use_gpu=True` and this class couldn't import cupy then it raise an error...

Reply:
Interesting. I can change ValueError to say "Please install cupy if you want to use gpu zoom".